nonebot_plugin_guild_patch/models.py

```diff
@@ -6,43 +7,130 @@
     Message,
     MessageEvent,
     NoticeEvent,
+    MessageSegment
 )
 from nonebot.log import logger
-from pydantic import BaseModel, Field, parse_obj_as, validator
+from nonebot.typing import overrides
+from nonebot.utils import escape_tag
+from pydantic import BaseModel, Field, parse_obj_as, validator, root_validator
 from typing_extensions import Literal
 
+
 Event_T = TypeVar("Event_T", bound=Type[Event])
 
 
 def register_event(event: Event_T) -> Event_T:
     Adapter.add_custom_model(event)
     logger.opt(colors=True).trace(
-        f"Custom event <e>{event.__event__!r}</e> registered "
-        f"from class <g>{event.__qualname__!r}</g>"
+        f"Custom event <e>{event.__qualname__!r}</e> registered "
+        f"from module <g>{inspect.getmodule(event).__name__!r}</g>"
     )
     return event
 
 
 @register_event
 class GuildMessageEvent(MessageEvent):
-    __event__ = "message.guild"
-    self_tiny_id: int
-
+    """收到频道消息"""
     message_type: Literal["guild"]
+    self_tiny_id: int
     message_id: str
     guild_id: int
     channel_id: int
 
     raw_message: str = Field(alias="message")
     font: None = None
 
-    @validator("raw_message", pre=True)
+    @validator('raw_message', pre=True)
     def _validate_raw_message(cls, raw_message):
         if isinstance(raw_message, str):
             return raw_message
         elif isinstance(raw_message, list):
             return str(parse_obj_as(Message, raw_message))
-        raise ValueError("unknown raw message type")
+        raise ValueError('unknown raw message type')
+
+    @root_validator(pre=False)
+    def _validate_is_tome(cls, values):
+        message = values.get('message')
+        self_tiny_id = values.get('self_tiny_id')
+        message, is_tome = cls._check_at_me(message=message, self_tiny_id=self_tiny_id)
+        values.update({'message': message, 'to_me': is_tome, 'raw_message': str(message)})
+        return values
+
+    @overrides(Event)
+    def is_tome(self) -> bool:
+        return self.to_me or any(
+            str(msg_seg.data.get('qq', '')) == str(self.self_tiny_id)
+            for msg_seg in self.message
+            if msg_seg.type == 'at'
+        )
+
+    @overrides(Event)
+    def get_event_description(self) -> str:
+        return (
+            f'Message {self.message_id} from {self.user_id}@[频道:{self.guild_id}/子频道:{self.channel_id}] "'
```

Comment:
```suggestion
            f'Message {self.message_id} from {self.user_id}@[Guild:{self.guild_id}/Channel:{self.channel_id}] "'
```
另外我记得这俩ID的位数都很长, 会不会有些刷屏?

Reply:
对比频道消息 log 和群消息 log 好像也没有长太多，应该还能接受

```
06-14 00:07:53 [SUCCESS] nonebot | ONEBOT V11 0000000000 | [message.guild.channel]: Message BABABAB/8888AAAAAAaaaaaAAAAAAAAaaA== from 123456789012345678@[频道:12345678901234567/子频道:1234567] "/萌图 艾雅法拉"
06-14 00:08:13 [SUCCESS] nonebot | ONEBOT V11 0000000000 | [message.group.normal]: Message 2055888888 from 1234567890@[群:123456789] "抱抱"
```

~~起码比原始的事件log要短~~